frame/support/src/storage/mod.rs

```diff
@@ -1010,6 +1013,121 @@ impl<T: Encode> StorageDecodeLength for Vec<T> {}
 /// format ever changes, we need to remove this here.
 impl<Hash: Encode> StorageAppend<DigestItem<Hash>> for Digest<Hash> {}
 
+/// Marker trait that will be implemented for types that support the `storage::try_append` api.
+///
+/// This trait is sealed.
+pub trait StorageTryAppend<Item>: StorageDecodeLength + private::Sealed {
```

Comment:
this is a new trait in order to be able to implement try_append for both WeakBoudedVec and BoundedVec without too much code duplication.